code/modules/mob/living/simple_animal/guardian/guardian_abilities.dm

```diff
@@ -0,0 +1,733 @@
+//SABAKU NO WA ORE STANDO DA
+
+/datum/guardian_abilities
+	var/toggle = FALSE
+	var/mob/living/simple_animal/hostile/guardian/stando = null
+	var/mob/living/user = null
+	var/cooldown = 0
+	var/value = 0 //VALUE SYSTEM:
+	var/battlecry = "ORA"
+	//basically, the total value of abilities a stand can have is 10, and more powerful abilities have more value, this means that
+	//randomized-ability stands can only have a set number of abilities both in quantity and quality, maintaining some form of balance.
+	//CRUCIAL: only randomized stands are to use the value system.
+
+/datum/guardian_abilities/proc/handle_stats()
+
+
+/datum/guardian_abilities/proc/life_act()
+
+
+/datum/guardian_abilities/proc/ability_act()
+
+
+/datum/guardian_abilities/proc/alt_ability_act() //ability to do on alt_click
+
+
+/datum/guardian_abilities/proc/handle_mode()
+
+
+/datum/guardian_abilities/proc/bump_reaction()
+
+
+/datum/guardian_abilities/proc/ranged_attack()
+
+
+/datum/guardian_abilities/proc/impact_act()
+
+
+/datum/guardian_abilities/proc/recall_act()
+
+
+/datum/guardian_abilities/proc/adjusthealth_act()
+
+
+/datum/guardian_abilities/proc/light_switch()
+
+
+/datum/guardian_abilities/proc/destroy_act()
+
+
+/datum/guardian_abilities/proc/manifest_act()
+
+
+/datum/guardian_abilities/proc/openfire_act()
+
+
+/datum/guardian_abilities/proc/move_act()
+
+
+/datum/guardian_abilities/proc/snapback_act()
+
+
+/datum/guardian_abilities/proc/boom_act()
+
+//ORA ORA ORA
+
+/datum/guardian_abilities/punch
+	value = 5
+
+
+/datum/guardian_abilities/punch/handle_stats()
+	stando.melee_damage_lower += 10
+	stando.melee_damage_upper += 10
+	stando.obj_damage = 80
+	stando.next_move_modifier = 0.8 //attacks 20% faster
+	stando.environment_smash = 2
+	stando.attack_sound = pick('sound/magic/standopunch.ogg', 'sound/magic/standopunch1.ogg', 'sound/magic/standopunch2.ogg')
```

Comment:
This won't randomize the attack sound multiple times, it needs to be in ability_act